### ISSUE TITLE
Add an exemption to annotation rule for single-line typealiases with single annotation

### ISSUE THREAD

**Issue by Kantis** [pinterest/ktlint#2152]
## Expected Behavior
When working with kotlinx-serialization one can create typealiases to declare which Serializer should be used for a type. Eg
```kotlin
typealias BigDecimalAsString = @Serializable(BigDecimalAsStringSerializer::class) BigDecimal
```

It would (imo) be nice to keep this formatted as the example above

## Current Behavior
Ktlint forces the formatting to be the following which IMO is less readable.
```kotlin
typealias BigDecimalAsString = 
    @Serializable(BigDecimalAsStringSerializer::class) 
    BigDecimal
```

## Additional information
* Current version of ktlint: 0.50.0

**Comment by Kantis** [comment]
note that the formatted variant also clashes with IntelliJs default formatting rules
<img width="500" alt="image" src="https://github.com/pinterest/ktlint/assets/316857/0fe034ce-ea28-432e-9bf0-b7f062b40b57">

**Comment by Kantis** [comment]
I was using the Ktlint IntelliJ plugin which is using a slightly older version of Ktlint. That version wanted it formatted as
```kotlin
typealias BigDecimalAsString = 
@Serializable(BigDecimalAsStringSerializer::class) 
BigDecimal
```
which imo wasn't very nice. The 0.50.0-way is good though, imo. Closing this.
